src/components/AddItems.jsx

```diff
@@ -0,0 +1,102 @@
+import { useCallback, useState } from 'react';
+import { useStateWithStorage } from '../utils';
+import { addItem } from '../api';
+
+const soonDate = 7;
+const kindOfSoonDate = 14;
+const notSoonDate = 30;
+
+export function AddItems() {
+	const [daysUntilNextPurchase, setDaysUntilNextPurchase] = useState(null);
+	const [itemName, setItemName] = useState('');
```

Comment:
Instead of using a state hold the `itemName`, which triggers a re-render each time the value changes, I wonder if we could extract the name directly from the form on submission? Since the submit handler is where all the validation is taking place anyway, this might help minimize unnecessary re-renders! 🚀